src/basic/util.c

```diff
@@ -832,6 +832,62 @@ uint64_t physical_memory_scale(uint64_t v, uint64_t max) {
         return r;
 }
 
+uint64_t system_tasks_max(void) {
+
+#if SIZEOF_PID_T == 4
+#define TASKS_MAX ((uint64_t) (INT32_MAX-1))
+#elif SIZEOF_PID_T == 2
+#define TASKS_MAX ((uint64_t) (INT16_MAX-1))
+#else
+#error "Unknown pid_t size"
+#endif
```

Comment:
i figure you mean `1 <<…` instead of `2 <<…` ?

yeah, this indeed might cause an overvflow. After all on 32bit systems size_t and pid_t are 32bit. Hence we'd calculate `1 << 4*8 = 1 << 32` which doesn't fit into 32bit anymore...

I thought about this a bit, but I figured it's more obvious if we just do it with the #if check and hence can be sure we won#t run into overflow issues...